src/core/Akka.Streams.Tests/Dsl/FramingSpec.cs

```diff
@@ -240,43 +241,98 @@ private static string RandomString(int length)
 
         private static readonly List<int> FieldOffsets = new List<int> {0, 1, 2, 3, 15, 16, 31, 32, 44, 107};
 
-        private static ByteString Encode(ByteString payload, int fieldOffset, int fieldLength, ByteOrder byteOrder)
+        private static ByteString Encode(ByteString payload, int fieldOffset, int fieldLength, ByteOrder byteOrder) =>
+            EncodeComplexFrame(payload, fieldLength, byteOrder, ByteString.FromBytes(new byte[fieldOffset]), ByteString.Empty);
+
+        private static ByteString EncodeComplexFrame(
+            ByteString payload, 
+            int fieldLength, 
+            ByteOrder byteOrder, 
+            ByteString offset, 
+            ByteString tail)
         {
             var h = ByteString.FromBytes(new byte[4].PutInt(payload.Count, order: byteOrder));
```

Comment:
With the `ByteStringBuilder` I bet we could also save on allocations here too...